runner/src/elf2tab.rs

```diff
@@ -48,7 +75,11 @@ pub fn convert_elf(cli: &Cli) -> OutFiles {
         "-o".as_ref(), tab_path.as_os_str(),
         "--protected-region-size".as_ref(), protected_size.as_ref(),
         "--stack".as_ref(), stack_size.as_ref(),
-        elf,
+        format!("{}{}", elf.to_str().unwrap(), if let Some(ref architecture) = architecture { 
+                format!(",{architecture}") 
+            } else { 
+                String::from("") 
+            }).as_ref(),
```

Comment:
This `format!` call should replace:

```rust
let elf = cli.elf.as_os_str();
```

above. That way, this debug block:

```rust
    if cli.verbose {
        println!("ELF file: {:?}", elf);
        println!("TBF path: {}", tbf_path.display());
    }
```

will print the correct value.

Reply:
I don't understand. The elf's and tbf's file names are printed correctly.

Reply:
Oh I see: the ELF file isn't moved, this is just a combination elf + architecture specification.

I'm still concerned about how difficult it is to read the `command.args()` call with the `format!()` call inside it. Do we need to support unknown boards? I think the answer is "no". If you make `get_platform_architecture` return `&'static str` and panic if it doesn't recognize the board, then the `format!()` call will become a single-line expression. I think that will make it more readable.